### PR TITLE
feat(ui5-checkbox): add value property

### PR DESCRIPTION
**Pull request by kineticjs**
Fixes: https://github.com/SAP/ui5-webcomponents/issues/11607

### CODE REVIEW

File: packages/main/cypress/specs/FormSupport.cy.tsx [hunk 1]
```diff
@@ -64,7 +65,7 @@ describe("Form support", () => {
 			.then($el => {
 				return getFormData($el.get(0));
 			})
-			.should("be.equal", "checkbox4=on&checkbox5=on");
+			.should("be.equal", "checkbox4=on&checkbox5=on&checkbox6=checkbox6Value");
```
[the remark this answers is not in the document]
Comment (kineticjs) [reply packages/main/cypress/specs/FormSupport.cy.tsx:68]: Dismissing this comment because:
checkbox5 is indeed initially unchecked 
but 
is checked later:
at line 55 
with: cy.get("#cb5").realClick();

So the final form data will include the value of checkbox5

File: packages/main/src/CheckBoxTemplate.tsx [hunk 2]
```diff
@@ -49,6 +49,7 @@ export default function CheckBoxTemplate(this: CheckBox) {
 				id={`${this._id}-CB`}
 				type="checkbox"
 				checked={this.checked}
+				value={this.value}
```
[the remark this answers is not in the document]
Comment (kineticjs) [reply packages/main/src/CheckBoxTemplate.tsx:52]: in testing we see that the template renderer *already* omits the `value` attribute if a default `value` of 'on' is being passed 
=> this why I'll propose to keep it short , i.e. 
value={this.value}

this way we also avoid mentioning the default literal value in more than one place in the code (i.e. repetition of the default value literal)

File: packages/main/cypress/specs/CheckBox.cy.tsx [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+			.find('[type="checkbox"]')
+			.should('have.attr', 'value', 'testValue');
+	});
+
+	it('should omit value attribute in shadow dom when checkbox has default value', () => {
+		cy.mount(<CheckBox />);
+		cy.get("[ui5-checkbox]")
+			.shadow()
+			.find('[type="checkbox"]')
+			.should('not.have.attr', 'value');
```
Comment (plamenivanov91) [inline packages/main/cypress/specs/CheckBox.cy.tsx:17]: If I revert the code in the Checkbox.ts to

	get formFormattedValue() {
		return this.checked ? "on" : null;
	}

this test still passes? Is it possible to alter the test in a way where is secures the newly added logic specifically ?

Comment (kineticjs) [reply packages/main/cypress/specs/CheckBox.cy.tsx:17]: yes, if we revert as you suggested, this particular test will still work... I intended with this test to check the default behavior (which is the same as before this change) and also to test that no **empty** `value` attribute will be rendered if no 1value1 was specified by the app.

So only the above test (i.e. the test named 'should have correct type and value attributes in shadow dom') will fail if we revert the code ..

The main test is in FormSupport.cy.tsx  (testing the scenario which required the addition of the `value` property).

I pondered on a way to edit the test you marked too, but no good idea so far... Do you have a particular scenario in mind?

**Review by plamenivanov91 — APPROVED** [review]
